docs/reference/outlets.md

```diff
@@ -0,0 +1,179 @@
+
+
+```html
+<div data-controller="search" data-search-result-outlet=".result"></div>
+```
+
+## Definitions
+
+Define controller identifiers in your controller class using the `static outlets` array. This array declares which other controller identifiers can be used as outlets on this controller:
+
+<meta data-controller="callout" data-callout-text-value='static outlets'>
+<meta data-controller="callout" data-callout-text-value='"result"'>
+
+
+```js
+// search_controller.js
+
+export default class extends Controller {
+  static outlets = [ "result" ]
+
+  connect () {
+    this.resultOutlets.forEach(result => ...)
+  }
+}
+```
+
+## Properties
+
+For each outlet defined in the `static outlets` array, Stimulus adds five properties to your controller, where `[name]` corresponds to the outlet's controller identifier:
+
+| Kind | Property name | Return Type | Effect
+| ---- | ------------- | ----------- | -----------
+| Existential | `has[Name]Outlet` | `Boolean` | Tests for presence of a `[name]` outlet
+| Singular | `[name]Outlet` | `Controller` | Returns the `Controller` instance of the first `[name]` outlet or throws an exception if none is present
+| Plural | `[name]Outlets` | `Array<Controller>` | Returns the `Controller` instances of all `[name]` outlets
+| Singular | `[name]OutletElement` | `Element` | Returns the Controller `Element` of the first `[name]` outlet or throws an exception if none is present
+| Plural | `[name]OutletElements` | `Array<Element>` | Returns the Controller `Element`'s of all `[name]` outlets
+
+## Accessing controllers and elements
+
+Since you get back a `Controller` instance from the `[name]Outlet` and `[name]Outlets` properties you are also able to access the Values, Classes, Targets and all of the other properties and functions that controller instance defines:
+
+```js
+this.resultOutlet.idValue
+this.resultOutlet.imageTarget
+this.resultOutlet.activeClasses
+```
+
+You are also able to invoke any function the outlet controller may define:
+
+```js
+// result_controller.js
+
+export default class extends Controller {
+  markAsSelected(event) {
+    // ...
+  }
+}
+
+// search_controller.js
+
+export default class extends Controller {
+  static outlets = [ "result" ]
+
+  selectAll(event) {
+    this.resultOutlets.forEach(result => result.markAsSelected(event))
+  }
+}
+```
+
+Similarly with the Outlet Element, it allows you to call any function or property on [`Element`](https://developer.mozilla.org/en-US/docs/Web/API/Element):
+
+```js
+this.resultOutletElement.dataset.value
+this.resultOutletElement.getAttribute("id")
+this.resultOutletElements.map(result => result.hasAttribute("selected"))
+```
+
+## Outlet callbacks
+
+Outlet callbacks are specially named functions called by Stimulus to let you respond to whenever an outlet is added or removed from the page.
+
+To observe outlet changes, define a function named `[name]OutletConnected()` or `[name]OutletDisconnected()`.
```

Comment:
```suggestion
To observe outlet changes, define a class method named `[name]OutletConnected()` or `[name]OutletDisconnected()`.
```

Not sure if needed but it may be clearer if we say that this is a class method?

Reply:
Thanks for the review!

> Some general comments below but what happens if you have an outlet that has two or more Controllers on it?

An element with two controllers attached to it also has two different controllers instances. You'll get the controller instance for the controller identifier you referenced as the outlet. 

> Maybe I missed something but does the name of the outlet have to match the name of the expected outlet controller?

Yes, the outlet name has to match the controller identifier. This is what's used to get right controller instance for a given element. 
The reason why you are able to provide a CSS selector is that you can specify which element/instance you want to associate with the source controller, because usually there are more than one instance of the same controller on the page.

Reply:
you are right, in the `targets` section they are also called "methods" 👍🏼